src/silverlabnwb/timings.py

```diff
@@ -0,0 +1,68 @@
+import abc
+
+import numpy as np
+import pandas as pd
+
+
+class LabViewTimings(metaclass=abc.ABCMeta):
+
+    def __init__(self):
+        self.pixel_time_offsets = None
+        self.cycle_time = None
+
+    @abc.abstractmethod
+    def _read_relative_times_file(self, file_path):
+        pass
+
+
+class LabViewTimingsPre2018(LabViewTimings):
+
+    def __init__(self, relative_times_path):
+        self._read_relative_times_file(relative_times_path)
+
+    def _read_relative_times_file(self, file_path):
+        raw_data = pd.read_csv(file_path, names=('RelativeTime', 'CycleTime'),
+                               sep='\t', dtype=np.float64) / 1e6
+        self.pixel_time_offsets = raw_data['RelativeTime']
+        self.cycle_time = raw_data['CycleTime'][0]
+
+
+class LabViewTimings231(LabViewTimings):
+
+    def __init__(self, relative_times_path, roi_path, n_cycles_per_trial, n_trials):
+        self._read_relative_times_file(relative_times_path)
+        self._format_pixel_time_offsets(roi_path, n_cycles_per_trial, n_trials)
+
+    def _read_relative_times_file(self, file_path):
+        raw_data = pd.read_csv(file_path, sep='\t', dtype=np.float64) / 1e6
+        self.pixel_time_offsets = raw_data['Image Time [us]']
+        self.pixel_time_offsets = self.pixel_time_offsets[self.pixel_time_offsets != 0]
+
+    def _format_pixel_time_offsets(self, roi_path, n_cycles_per_trial, n_trials):
+        roi_data = pd.read_csv(roi_path, sep='\t', dtype=np.float64)
+        n_rois = len(roi_data['ROI index'])
+        # in the future, we can access more shape parameters here too, for variable size ROIs in those cases,
+        # we will need to access each individual row here separately. for now, take first row and all ROIs are same
+        # size and orientation. Also, we may need to store the angle somewhere in the future (to reconstruct the ROI
+        # in 3D, in which case the ==0 assertion may become imprecise and we would need a tolerance to compare with a
+        # double value.
```

Comment:
Good to have the future changes in mind! We  could use `math.isclose` or `np.isclose` for checking against 0 in that case.